fcrepo-persistence-common/src/main/java/org/fcrepo/persistence/common/ResourceHeaderUtils.java

```diff
@@ -111,6 +113,9 @@ public static void touchModificationHeaders(final ResourceHeadersImpl headers, f
         }
         headers.setLastModifiedDate(instant);
         headers.setLastModifiedBy(userPrincipal);
+
+        final String stateToken = DigestUtils.md5Hex(instant.toEpochMilli() + "").toUpperCase();
```

Comment:
I'm not 100% sure, but if this is casting the long to a String I think 
```
final String stateToken = DigestUtils.md5Hex(String.valueOf(instant.toEpochMilli())).toUpperCase();
```
would be clearer...unless it doesn't work as expected?

Reply:
I'm not sure if casting a long to a string will cause the result to include commas.

Reply:
I believe the "state token" should relate to the state of the resource, and therefore be reproducible in the case that the state of the resource does not change. 
The idea is that the state token can be subsequently used by a client when requesting to make a change to a resource. If the state token can not be reproduced on the server-side when determining whether the client-provided state token represents the current state or not... then the utility of the token is lost.

Reply:
I think this should be addressed in a separate JIRA: https://jira.lyrasis.org/browse/FCREPO-3169

Reply:
Either `String.valueOf` or `Long.toString` would be fine and not add commas, but they might not matter at this point

Reply:
Done.